File: RickLocalizationAPI/Domain/Models/RickModel.cs
using Domain.Models;
using System.Collections.Generic;

namespace RickLocalizationAPI.Models
{
    public class RickModel
    {
        public int Id { get; set; }

        public string Species { get; set; }

        public string Image { get; set; }

        public string Name { get; set; }

        public int Age { get; set; }

        public string Occupation { get; set; }

        public MortyModel Morty { get; set; }

        public DimensionModel Dimension { get; set; }

        public List<RickDimensionModel> RickDimensions { get; set; }

    }
}

File: RickLocalizationApp/src/app/models/rick.model.ts
import { DimensionModel } from "./dimension.model";
import { MortyModel } from "./morty.model";
import { RickDimension } from "./rick-dimension.model";

export class RickModel {
    id: number;
    species: string;
    image: string;
    name: string;
    age: number;
    occupation: string;
    morty: MortyModel;
    dimension: DimensionModel;
    // travels: RickDimension[];
  rickDimensions: RickDimension[];
}